ouroboros-consensus/ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Ledger/Address.hs

```diff
@@ -1,17 +1,37 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
 module Ouroboros.Consensus.Mock.Ledger.Address (
     Addr
   , AddrDist
   , mkAddrDist
   ) where
 
+import           Codec.Serialise (Serialise)
+import           Control.DeepSeq (NFData)
 import           Data.Map.Strict (Map)
 import qualified Data.Map.Strict as Map
+import           Data.String
+
+import           Cardano.Prelude (NoUnexpectedThunks)
 
 import           Ouroboros.Consensus.Node.ProtocolInfo
 import           Ouroboros.Consensus.NodeId (NodeId (..))
+import           Ouroboros.Consensus.Util.Condense
 
 -- | Mock address
-type Addr = String
+newtype Addr = Addr String
+  deriving (
+      Show
+    , Eq
+    , Ord
+    , IsString
+    , Serialise
+    , NFData
```

Comment:
Just wondering, where do we use the `NFData` instance?

Reply:
I think we force elements in the mock ledger rather than enforcing it in the types.